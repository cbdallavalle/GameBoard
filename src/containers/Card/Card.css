.Card {
  display: flex;
  flex-direction: row;
  position: relative;
  margin: 10px auto;
  height: 260px;
  width: 100%;
  background-color: #ffffffcf;
  box-sizing: border-box;
  padding: 15px;
  font-family: 'Open Sans', sans-serif; }

.games, .friends {
  animation: 0.7s cubic-bezier(0.29, 1.01, 0.45, 0.94) 0s 1 slideInFromLeft; }

@keyframes slideInFromLeft {
  0% {
    transform: translateX(-100%); }
  100% {
    transform: translateX(0); } }

.Card h3 {
  font-weight: bold;
  text-align: center; }

.Card img {
  margin-top: 10px;
  max-width: 180px; }

.info-div {
  width: 50px;
  height: 50px;
  background-color: magenta; }

.friends .game-info,
.games .game-info {
  width: 30%;
  line-height: 20px; }

.search #game-title {
  width: 30%; }

.search #game-description {
  width: 55%;
  line-height: 20px;
  padding: 0 40px 0 15px; }

.search #review-cont {
  display: none; }

#game-description {
  padding: 0 5px;
  width: 40%;
  height: 240px;
  overflow-y: scroll; }

#add,
#edit {
  position: absolute;
  top: 8px;
  right: 10px;
  width: 20px;
  height: 20px;
  border-radius: 10px;
  background-repeat: no-repeat;
  background-position: center; }
  #add:hover,
  #edit:hover {
    cursor: pointer; }

#add {
  background-image: url("../../assets/plus.svg"); }
  #add:hover {
    background-image: url("../../assets/plus-hover.svg"); }

#edit {
  background-image: url("../../assets/pencil.svg"); }
  #edit:hover {
    background-image: url("../../assets/pencil-hover.svg"); }

#delete, #add {
  cursor: pointer; }

#owned {
  width: 15px;
}

.search #edit,
.friends #edit {
  display: none; }

.games #add,
.search #add {
  display: none; }
