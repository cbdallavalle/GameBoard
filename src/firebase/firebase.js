import * as firebase from 'firebase';

const config = {
  apiKey: "AIzaSyBDU0_Pa8xdAWKmtKKfo3QkbrtR56cl024",
  authDomain: "dallavalle-gameboard.firebaseapp.com",
  databaseURL: "https://dallavalle-gameboard.firebaseio.com",
  projectId: "dallavalle-gameboard",
  storageBucket: "",
  messagingSenderId: "250758219123"
};

if(!firebase.apps.length) {
  firebase.initializeApp(config)
}

const auth = firebase.auth();

export {
  auth,
}