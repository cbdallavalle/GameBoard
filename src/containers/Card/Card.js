import React, { Component } from 'react';
import PropTypes from 'prop-types';
import deleteIcon from '../../assets/error.svg';
import addIcon from '../../assets/plus.svg';
import editIcon from '../../assets/pencil.svg';
import check from '../../assets/success.svg';
import negative from '../../assets/negative.svg';
import { updateFavorites } from '../../actions';
import { Link, Route } from 'react-router-dom';
import EditGame from '../../components/EditGame/EditGame';

import { connect } from 'react-redux';
import { db } from '../../firebase';
import './Card.css';

export class Card extends Component {
  constructor() {
    super();
    this.state = {
      error: '',
      edit: false,
      review: {}
    };
  }

  componentDidMount = () => {
    this.getGameReview();
  }

  friendsName = () => {
    return this.props.type === 'friends' 
      ? this.props.friendName + ' played:' 
      : null;
  }

  handleEdit = () => {
    this.setState({ edit: !this.state.edit })
  }

  handleAdd = async () => {
    try {
      await db.doAddFavoriteData(this.props.user.uid, this.props.favorite);
      const favorites = await db.getFavorites(this.props.user.uid);
      this.props.updateFavorites(favorites);
    } catch (error) {
      this.setState({ error: error.message });
    }
  }

  getGameReview = async() => {
    try {
      const review = await db.doGetReviewData(this.props.user.uid, this.props.favorite.name);
      review && this.setState({ review });
    } catch (error) {
      this.setState({ error: error.message });
    }
  }

  determineOwned = () => {
    return this.state.review.owned === 'yes' 
    ? <p id="owned-p">owned <i class="far fa-check-circle"></i></p>
    : <p id="owned-p">played <i class="far fa-play-circle"></i></p>
  }

  determineDisplay = () => {
    const { description, name, thumbnail } = this.props.favorite;
    const { rating, review, timesPlayed } = this.state.review;

    return !this.state.edit ?
    <article className={`Card ${this.props.type}`}>
        <div className="game-info" id="game-title">
          <h3>{ this.friendsName() }</h3>
          <h3>{ name }</h3>
          <img src={ thumbnail } alt="game-icon" />
          { this.determineOwned() }
          <div
            id="edit"
            alt="edit a game in your collection"
            onClick={this.handleEdit}
          >
          </div>
          <div
            id="add"
            alt="edit a game in your collection"
            onClick={this.handleAdd} 
          >
          </div>
        </div>
        <div className="game-info" id="game-description">
          <p>{ description }</p>
        </div>
        <div className={`game-info`} id="review-cont">
          <h3><span>{rating}</span>/5</h3>
          <p>
            { review } 
          </p>
        </div>
      </article>
      :
      <EditGame favorite={this.props.favorite} handleEdit={this.handleEdit} />
  }

  render() {
    return (
      <div>
      { this.determineDisplay() }
      </div>
    ); 
  }
}

Card.propTypes = {
  user: PropTypes.object.isRequired,
  favorite: PropTypes.object.isRequired,
  friendName: PropTypes.string,
  type: PropTypes.string,
  display: PropTypes.string,
  updateFavorites: PropTypes.func.isRequired
};

export const mapStateToProps = state => ({
  user: state.user
});

export const mapDispatchToProps = dispatch => ({
  updateFavorites: favorites => dispatch(updateFavorites(favorites))
});

export default connect(mapStateToProps, mapDispatchToProps)(Card);