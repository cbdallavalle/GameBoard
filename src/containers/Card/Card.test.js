/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import { Card, mapStateToProps, mapDispatchToProps } from './Card';
import { shallow } from 'enzyme';
import { db } from '../../firebase';
import { mockData } from '../../mockData/mockData';


describe('Card', () => {
  let wrapper;
  const mockUpdateFavorites = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <Card 
        user={{}}
        key={ 0 }
        favorite={ {} }
        friendName={ 'coda' }
        type={ 'friends' }
        updateFavorites={mockUpdateFavorites}
      />
    )
  })

  it('should match the snapshot', () => {
    const wrapper = shallow(
      <Card 
        user={{}}
        key={ 0 }
        favorite={ {} }
        friendName={ 'friend' }
        type={ 'games' }
        updateFavorites={mockUpdateFavorites}
      />
    )
    expect(wrapper).toMatchSnapshot();
  })

  it('should have a default state of contenteditable to be false, and error to be an empty string', () => {

    expect(wrapper.state()).toEqual({contenteditable: 'false', error: ''})
  })

  it('should setState so that contenteditable is true if the type is games', async() => {
    const wrapper = await shallow(
      <Card 
        user={{}}
        key={ 0 }
        favorite={ {} }
        friendName={ 'friend' }
        type={ 'games' }
        updateFavorites={mockUpdateFavorites}
      />
    )

    expect(wrapper.state()).toEqual({contenteditable: 'true', error: ''})
  })

  //friendsName
  it('friendsName should match the snapshot when a friends name is provided', () => {

    expect(wrapper).toMatchSnapshot();
  })

  it.skip('handleBlur should set state to have an error if error is caught', async() => {

    window.fetch = jest.fn().mockImplementation( () => Promise.resolve({
      status: 500,
      error: new Error('unable to load friends')
    }))

    const mockCardEvent = {target: {innerHTML: 'blahh'}}

    await wrapper.instance().handleBlur(mockCardEvent);
    expect(wrapper.state().error).toEqual('error')
  })

  it('handleBlur should call doWriteReviewData', () => {
    db.doWriteReviewData = jest.fn();
    const mockCardEvent = {target: {innerHTML: 'blahh'}}

    wrapper.instance().handleBlur(mockCardEvent);
    expect(db.doWriteReviewData).toHaveBeenCalled();
  })

  it('handleDelete should set state with an error if an error is caught', async() => {

    await wrapper.instance().handleDelete();
    expect(wrapper.state().error).toEqual('unable to delete')
  })

  it('handleDelete should call two db functions and updateFavorites', async() => {
    db.doDeleteFavoriteData = jest.fn();
    db.getFavorites = () => mockData.mockFavorites
    
    await wrapper.instance().handleDelete();
    expect(mockUpdateFavorites).toHaveBeenCalledWith(mockData.mockFavorites)
  })

  it('handleAdd should call set state with an error if error is caught', async() => {
    await wrapper.instance().handleAdd();
    expect(wrapper.state().error).toEqual('unable to add favorite data')
  })

  it('handleAdd should call two db functions and updateFavorites', async() => {
    db.doAddFavoriteData = jest.fn();
    db.getFavorites = () => mockData.mockFavorites

    const mockCardEvent = {target: {innerHTML: 'blahh'}}

    await wrapper.instance().props.updateFavorites.mockClear();
    await wrapper.instance().handleAdd(mockCardEvent);
    expect(mockUpdateFavorites).toHaveBeenCalledWith(mockData.mockFavorites)
  })
})

describe('mapStateToProps', () => {
  it('should take in state and return a user object', () => {
    expect(mapStateToProps(mockData.mockMSTPUserState)).toEqual(mockData.mockMSTPUser)
  })
})

describe('mapDispatchToProps', () => {
  it('should call mockDispatch on updateFavorites', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);

    mapped.updateFavorites();
    expect(mockDispatch).toHaveBeenCalled();
  })
})