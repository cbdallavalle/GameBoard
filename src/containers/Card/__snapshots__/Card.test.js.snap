// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card friendsName should match the snapshot when a friends name is provided 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    favorite={Object {}}
    friendName="coda"
    type="friends"
    updateFavorites={[Function]}
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <article
        className="Card friends"
>
        <div
                className="game-info"
                id="game-title"
        >
                <h3>
                        coda played:
                </h3>
                <h3 />
                <img
                        alt="game-icon"
                        src={undefined}
                />
                <p
                        id="owned-p"
                >
                        played 
                        <i
                                class="far fa-play-circle"
                        />
                </p>
                <div
                        alt="edit a game in your collection"
                        id="edit"
                        onClick={[Function]}
                />
                <div
                        alt="edit a game in your collection"
                        id="add"
                        onClick={[Function]}
                />
        </div>
        <div
                className="game-info"
                id="game-description"
        >
                <p />
        </div>
        <div
                className="game-info"
                id="review-cont"
        >
                <h3>
                        <span />
                        /5
                </h3>
                <p />
        </div>
</article>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="game-info"
            id="game-title"
>
            <h3>
                        coda played:
            </h3>
            <h3 />
            <img
                        alt="game-icon"
                        src={undefined}
            />
            <p
                        id="owned-p"
            >
                        played 
                        <i
                                    class="far fa-play-circle"
                        />
            </p>
            <div
                        alt="edit a game in your collection"
                        id="edit"
                        onClick={[Function]}
            />
            <div
                        alt="edit a game in your collection"
                        id="add"
                        onClick={[Function]}
            />
</div>,
          <div
            className="game-info"
            id="game-description"
>
            <p />
</div>,
          <div
            className="game-info"
            id="review-cont"
>
            <h3>
                        <span />
                        /5
            </h3>
            <p />
</div>,
        ],
        "className": "Card friends",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                coda played:
</h3>,
              <h3 />,
              <img
                alt="game-icon"
                src={undefined}
/>,
              <p
                id="owned-p"
>
                played 
                <i
                                class="far fa-play-circle"
                />
</p>,
              <div
                alt="edit a game in your collection"
                id="edit"
                onClick={[Function]}
/>,
              <div
                alt="edit a game in your collection"
                id="add"
                onClick={[Function]}
/>,
            ],
            "className": "game-info",
            "id": "game-title",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "coda played:",
              },
              "ref": null,
              "rendered": "coda played:",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "game-icon",
                "src": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "played ",
                  <i
                    class="far fa-play-circle"
/>,
                ],
                "id": "owned-p",
              },
              "ref": null,
              "rendered": Array [
                "played ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "class": "far fa-play-circle",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "edit a game in your collection",
                "id": "edit",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "edit a game in your collection",
                "id": "add",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p />,
            "className": "game-info",
            "id": "game-description",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                <span />
                /5
</h3>,
              <p />,
            ],
            "className": "game-info",
            "id": "review-cont",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span />,
                  "/5",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                "/5",
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <article
          className="Card friends"
>
          <div
                    className="game-info"
                    id="game-title"
          >
                    <h3>
                              coda played:
                    </h3>
                    <h3 />
                    <img
                              alt="game-icon"
                              src={undefined}
                    />
                    <p
                              id="owned-p"
                    >
                              played 
                              <i
                                        class="far fa-play-circle"
                              />
                    </p>
                    <div
                              alt="edit a game in your collection"
                              id="edit"
                              onClick={[Function]}
                    />
                    <div
                              alt="edit a game in your collection"
                              id="add"
                              onClick={[Function]}
                    />
          </div>
          <div
                    className="game-info"
                    id="game-description"
          >
                    <p />
          </div>
          <div
                    className="game-info"
                    id="review-cont"
          >
                    <h3>
                              <span />
                              /5
                    </h3>
                    <p />
          </div>
</article>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="game-info"
              id="game-title"
>
              <h3>
                            coda played:
              </h3>
              <h3 />
              <img
                            alt="game-icon"
                            src={undefined}
              />
              <p
                            id="owned-p"
              >
                            played 
                            <i
                                          class="far fa-play-circle"
                            />
              </p>
              <div
                            alt="edit a game in your collection"
                            id="edit"
                            onClick={[Function]}
              />
              <div
                            alt="edit a game in your collection"
                            id="add"
                            onClick={[Function]}
              />
</div>,
            <div
              className="game-info"
              id="game-description"
>
              <p />
</div>,
            <div
              className="game-info"
              id="review-cont"
>
              <h3>
                            <span />
                            /5
              </h3>
              <p />
</div>,
          ],
          "className": "Card friends",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  coda played:
</h3>,
                <h3 />,
                <img
                  alt="game-icon"
                  src={undefined}
/>,
                <p
                  id="owned-p"
>
                  played 
                  <i
                                    class="far fa-play-circle"
                  />
</p>,
                <div
                  alt="edit a game in your collection"
                  id="edit"
                  onClick={[Function]}
/>,
                <div
                  alt="edit a game in your collection"
                  id="add"
                  onClick={[Function]}
/>,
              ],
              "className": "game-info",
              "id": "game-title",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "coda played:",
                },
                "ref": null,
                "rendered": "coda played:",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "game-icon",
                  "src": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "played ",
                    <i
                      class="far fa-play-circle"
/>,
                  ],
                  "id": "owned-p",
                },
                "ref": null,
                "rendered": Array [
                  "played ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "class": "far fa-play-circle",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "edit a game in your collection",
                  "id": "edit",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "edit a game in your collection",
                  "id": "add",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p />,
              "className": "game-info",
              "id": "game-description",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  <span />
                  /5
</h3>,
                <p />,
              ],
              "className": "game-info",
              "id": "review-cont",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span />,
                    "/5",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "span",
                  },
                  "/5",
                ],
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "article",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    favorite={Object {}}
    friendName="friend"
    type="games"
    updateFavorites={[Function]}
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <article
        className="Card games"
>
        <div
                className="game-info"
                id="game-title"
        >
                <h3 />
                <h3 />
                <img
                        alt="game-icon"
                        src={undefined}
                />
                <p
                        id="owned-p"
                >
                        played 
                        <i
                                class="far fa-play-circle"
                        />
                </p>
                <div
                        alt="edit a game in your collection"
                        id="edit"
                        onClick={[Function]}
                />
                <div
                        alt="edit a game in your collection"
                        id="add"
                        onClick={[Function]}
                />
        </div>
        <div
                className="game-info"
                id="game-description"
        >
                <p />
        </div>
        <div
                className="game-info"
                id="review-cont"
        >
                <h3>
                        <span />
                        /5
                </h3>
                <p />
        </div>
</article>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="game-info"
            id="game-title"
>
            <h3 />
            <h3 />
            <img
                        alt="game-icon"
                        src={undefined}
            />
            <p
                        id="owned-p"
            >
                        played 
                        <i
                                    class="far fa-play-circle"
                        />
            </p>
            <div
                        alt="edit a game in your collection"
                        id="edit"
                        onClick={[Function]}
            />
            <div
                        alt="edit a game in your collection"
                        id="add"
                        onClick={[Function]}
            />
</div>,
          <div
            className="game-info"
            id="game-description"
>
            <p />
</div>,
          <div
            className="game-info"
            id="review-cont"
>
            <h3>
                        <span />
                        /5
            </h3>
            <p />
</div>,
        ],
        "className": "Card games",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3 />,
              <h3 />,
              <img
                alt="game-icon"
                src={undefined}
/>,
              <p
                id="owned-p"
>
                played 
                <i
                                class="far fa-play-circle"
                />
</p>,
              <div
                alt="edit a game in your collection"
                id="edit"
                onClick={[Function]}
/>,
              <div
                alt="edit a game in your collection"
                id="add"
                onClick={[Function]}
/>,
            ],
            "className": "game-info",
            "id": "game-title",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": null,
              },
              "ref": null,
              "rendered": null,
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "game-icon",
                "src": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "played ",
                  <i
                    class="far fa-play-circle"
/>,
                ],
                "id": "owned-p",
              },
              "ref": null,
              "rendered": Array [
                "played ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "class": "far fa-play-circle",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "edit a game in your collection",
                "id": "edit",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "edit a game in your collection",
                "id": "add",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p />,
            "className": "game-info",
            "id": "game-description",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                <span />
                /5
</h3>,
              <p />,
            ],
            "className": "game-info",
            "id": "review-cont",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span />,
                  "/5",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                "/5",
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <article
          className="Card games"
>
          <div
                    className="game-info"
                    id="game-title"
          >
                    <h3 />
                    <h3 />
                    <img
                              alt="game-icon"
                              src={undefined}
                    />
                    <p
                              id="owned-p"
                    >
                              played 
                              <i
                                        class="far fa-play-circle"
                              />
                    </p>
                    <div
                              alt="edit a game in your collection"
                              id="edit"
                              onClick={[Function]}
                    />
                    <div
                              alt="edit a game in your collection"
                              id="add"
                              onClick={[Function]}
                    />
          </div>
          <div
                    className="game-info"
                    id="game-description"
          >
                    <p />
          </div>
          <div
                    className="game-info"
                    id="review-cont"
          >
                    <h3>
                              <span />
                              /5
                    </h3>
                    <p />
          </div>
</article>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="game-info"
              id="game-title"
>
              <h3 />
              <h3 />
              <img
                            alt="game-icon"
                            src={undefined}
              />
              <p
                            id="owned-p"
              >
                            played 
                            <i
                                          class="far fa-play-circle"
                            />
              </p>
              <div
                            alt="edit a game in your collection"
                            id="edit"
                            onClick={[Function]}
              />
              <div
                            alt="edit a game in your collection"
                            id="add"
                            onClick={[Function]}
              />
</div>,
            <div
              className="game-info"
              id="game-description"
>
              <p />
</div>,
            <div
              className="game-info"
              id="review-cont"
>
              <h3>
                            <span />
                            /5
              </h3>
              <p />
</div>,
          ],
          "className": "Card games",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3 />,
                <h3 />,
                <img
                  alt="game-icon"
                  src={undefined}
/>,
                <p
                  id="owned-p"
>
                  played 
                  <i
                                    class="far fa-play-circle"
                  />
</p>,
                <div
                  alt="edit a game in your collection"
                  id="edit"
                  onClick={[Function]}
/>,
                <div
                  alt="edit a game in your collection"
                  id="add"
                  onClick={[Function]}
/>,
              ],
              "className": "game-info",
              "id": "game-title",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": null,
                },
                "ref": null,
                "rendered": null,
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "game-icon",
                  "src": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "played ",
                    <i
                      class="far fa-play-circle"
/>,
                  ],
                  "id": "owned-p",
                },
                "ref": null,
                "rendered": Array [
                  "played ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "class": "far fa-play-circle",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "edit a game in your collection",
                  "id": "edit",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "edit a game in your collection",
                  "id": "add",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p />,
              "className": "game-info",
              "id": "game-description",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  <span />
                  /5
</h3>,
                <p />,
              ],
              "className": "game-info",
              "id": "review-cont",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span />,
                    "/5",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "span",
                  },
                  "/5",
                ],
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "article",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
